prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,23 +86,338 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
+        logger.info("Processor data: %s", processor_data)
+        return processor_count, processor_data
+
+    def get_storage_controller_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
+        """Return storage controller data and count.
+
+        Returns:
+            storage_controller_count: Dict of system ids mapped to the total number
+            of storage controllers on that system.
+            storage_controller_data: Dict of system ids mapped to list of storage
+            controllers on that system.
+        """
+        storage_controller_data: Dict[str, List] = {}
+        # storage controllers on each system
+        storage_controller_count: Dict[str, int] = {}
+        storage_root_uri_pattern = "/redfish/v1/Systems/{}/Storage"
+        redfish_obj: Optional[HttpClient] = self._get_redfish_obj()
+
+        if redfish_obj:
+            try:
+                system_ids: List[str] = redfish_utilities.systems.get_system_ids(redfish_obj)
+            except redfish_utilities.systems.RedfishSystemNotFoundError:
+                logger.info("No system instances found.")
+                return storage_controller_count, storage_controller_data
+
+            for system_id in system_ids:
+                storage_controller_count[system_id] = 0
+
+                # List of storage ids
+                storage_ids: List[str] = redfish_utilities.collections.get_collection_ids(
+                    redfish_obj, storage_root_uri_pattern.format(system_id)
+                )
+                storage_controller_data_in_curr_system = []
+                for storage_id in storage_ids:
+                    # eg: /redfish/v1/Systems/1/Storage/XYZ123
+                    curr_storage_uri = (
+                        storage_root_uri_pattern.format(system_id) + "/" + storage_id
+                    )
+
+                    # list of storage controllers for that storage id
+                    storage_controllers_list: List[Dict] = redfish_obj.get(curr_storage_uri).dict[
+                        "StorageControllers"
+                    ]
+                    storage_controller_count[system_id] += len(storage_controllers_list)
+
+                    # picking out the required data from each storage controller in the list
+                    for data in storage_controllers_list:
+                        storage_controller_data_in_curr_system.append(
+                            {
+                                "storage_id": storage_id,
+                                "controller_id": data["MemberId"],
+                                "state": data["Status"]["State"],
+                                "health": data["Status"]["Health"] or "NA",
+                            }
+                        )
+
+                storage_controller_data[system_id] = storage_controller_data_in_curr_system
+
+            redfish_obj.logout()
+
+        logger.info("storage controller count: %s", storage_controller_count)
```

Comment:
**nitpick:** 

This shouldn't be info level log.

Reply:
Will change it to debug level log :+1: